packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -151,22 +154,44 @@ const NpsSurvey = () => {
     return null;
   }
 
-  const handleSubmitResponse = () => {
-    setNpsSurveySettings((settings) => ({
-      ...settings,
-      lastResponseDate: new Date(),
-      firstDismissalDate: null,
-      lastDismissalDate: null,
-    }));
-    // TODO: send response to the backend
-
-    // if success show thank you message
-    setIsFeedbackResponse(true);
+  const handleSubmitResponse = async ({ npsSurveyRating: rating, npsSurveyFeedback: comment }) => {
+    setIsSendingSurvey(true);
+    try {
+      const { email } = auth.getUserInfo();
+
+      const res = await fetch('https://analytics.strapi.io/submit-nps', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, rating, comment }),
+      });
+
+      if (!res.ok) {
+        throw new Error('Failed to submit NPS survey');
+      }
 
-    // Thank you message displayed in the banner should disappear after few seconds.
-    setTimeout(() => {
+      setNpsSurveySettings((settings) => ({
+        ...settings,
+        lastResponseDate: new Date(),
+        firstDismissalDate: null,
+        lastDismissalDate: null,
+      }));
+
+      setIsFeedbackResponse(true);
+
+      // Thank you message displayed in the banner should disappear after few seconds.
+      setTimeout(() => {
+        setSurveyIsShown(false);
+      }, 3000);
+    } catch {
+      toggleNotification({
+        type: 'warning',
+        message: formatMessage({ id: 'notification.error' }),
+      });
```

Comment:
I'm wondering if this is really what we want to do, because the user doesn't have a chance to fix the error (probably). So the error notification might cause more confusion than harm?

We could say the request has failed or something like that to motivate the user to send it again? Not sure ...

Reply:
Yeah I think that would make sense from a users pov.